### ISSUE TITLE
Accessibility audit

### ISSUE THREAD

**Issue by AlexITC** [wiringbits/scala-webapp-template#433]
It would be super helpful to get an idea on which parts of the webapp need to be improved in order to make the application accessible for visually impaired people.

The end goal would be to eventually make accessibility a standard feature for our template.

**Comment by ayushkoli772** [comment]
Hello @AlexITC,

I'm very interested in contributing to this issue. I have previous experience working on projects involving visually impaired users,  where I gained an understanding of the varying levels and types of impairment. While I'm relatively new to the Scala environment, I'm very eager to learn and contribute to this issue. I'd appreciate it if this issue could be assigned to me.

**Comment by ayushkoli772** [comment]
Hey @AlexITC,

I have thoroughly reviewed the web app to check its accessibility for visually impaired users. While the web app is largely compatible with commonly used screen readers, here's a summary of my findings and suggested improvements:

1. Proper HTML Structure and Labeling: The current HTML structure supports screen readers, ensuring proper navigation and comprehension. **No issues found here**.

2. Responsive Website: Although the web app is responsive, I noticed an issue regarding the toggle menu button's readability on smaller screens. This **button is not easily identifiable by screen readers**, potentially obstructing navigation for visually impaired users on mobile devices.

3. Audio Feedback Options: To improve the completion process, **introducing audio feedback** options during the completion process can significantly benefit users reliant on screen readers.

4. Contrast and Font Size for Warnings: Increasing contrast colors and font sizes **specifically for warning messages** would enhance readability and ensure crucial information is more perceivable for visually impaired users.

5. Keyboard Navigation: Verified that the web app supports proper keyboard navigation. **No issues found here**.

6. Screen Reader Compatibility: While the web app is generally compatible with screen readers, I observed a need for enhanced alternative text for images. **Providing descriptive alternative text** would significantly improve the understanding of visual content for users relying on screen readers. 

These suggestions aim to further enhance the accessibility of our web app for visually impaired users. I welcome any feedback or additional considerations regarding these proposed improvements. Thank you

**Comment by AlexITC** [comment]
Thanks for sharing, are you interested interesting in working on any of these improvements?

**Comment by ayushkoli772** [comment]
Yes, I'm interested in working on these improvements but I lack Scala.js knowledge so first I'd like to take time to familiarize myself with Scala before diving into addressing these issues. I'll keep you updated on my progress and I will start contributing as soon as i feel sufficiently confident in Scala.js.

Thank you for your understanding and patience.

**Comment by ayushkoli772** [comment]
Hey @AlexITC, I gave few days to learn Scala and now Im moving forward to get started with installing webapp locally and saw the video but it was for Ubuntu and currently i use windows 10 OS. Is there any other resource i can refer to or there will be no difference even if i follow same step for windows?

Thank you for your time :)

**Comment by AlexITC** [comment]
The steps likely work if you use WSL, otherwise, the stuff to install is more-or-less the same but the approach can be different.

Feel free to raise any issues you find.

Thanks.

**Comment by ayushkoli772** [comment]
Got it. I will update you about my progress.

Thank you for your time :)

**Comment by AlexITC** [comment]
Just follow the "Compile-time dependencies" section, then, run this:
- `API_URL="https://template-demo.wiringbits.net/api" sbt dev-web`

**Comment by AlexITC** [comment]
Unfortunately, I can't help you with the Windows setup, given that you don't need postgres, have you tried setting everything up with Windows powershell?

Related to using a VM, that should not be an issue.

**Comment by AlexITC** [comment]
Scala-center is participating on GSoC 2024, I was invited to be a mentor but I had to reject due to my limited time.

If you are interested, feel free to propose a project related to this template and I'll evaluate whether I can make the time to be a mentor or provide someone else from Wiringbits.

Good luck.

**Comment by AlexITC** [comment]
Great.

Feel free to create a new issue summarizing your findings.

**Comment by ayushkoli772** [comment]
Hello @AlexITC, I'm sincerely sorry for the delay in getting back to you about this issue. I encountered an unexpected issue with my development environment a few days ago. 

While troubleshooting an Out of Memory problem in VirtualBox, I made some changes to my Ubuntu VM instance. Unfortunately, these changes led to data loss, and I didn't have a recent backup, so I had to rebuild the entire environment from scratch.

I want to assure you that I'm actively working on the project now. I'll create a new issue and follow up with a PR as soon as I have something concrete to contribute.

Thanks for your understanding and patience :)
